finance-api/api-reference.md

```diff
@@ -0,0 +1,282 @@
+|totalBalance |integer |The account balance. It can be negative. For example, `-300`
+|currencyId |string |The currency of the account balance. For example, `SEK`.
+</details>
+
+
+## Fetch account transactions
+Returns all transactions or transactions of certain types from a merchant's Zettle preliminary or liquid account during a specific period.
+
+```
+GET /organizations/{organizationUuid}/accounts/{accountTypeGroup}/transactions?{start}&{end}
+```
+
+See example [Fetch transactions for a liquid account](#fetch-transactions-for-a-liquid-account).
+
+### Parameters
+
+<details open="true">
+<summary>Parameters</summary>
+
+|Name |Type |In |Required/Optional |Description
+|:---- |:---- |:---- |:---- |:----
+|organizationUuid |string |path |required |Unique identifier for your organization. You can specify the value with one of the following: <br/><ul><li> Use `self` as the value. This will retrieve your organizationUuid from the authentication token in the request.</li><li> Get it by using the https://oauth.izettle.com/users/me endpoint of OAuth2 API. See [OAuth2 API](https://github.com/iZettle/api-documentation/blob/master/authorization.adoc) for more information.</li></ul> 
+|accountTypeGroup |string |path |required |The account type from which the data like transactions is retrieved. You can use one of the following account types: <br/><ul><li> `PRELIMINARY` account where transactions are to be confirmed by third-party acquirers.</li><li> `LIQUID` account where transactions are to be paid out to the merchant.</li></ul>
+|start |string |query |required |A start time (inclusive) in the UTC time zone from when the transactions will be fetched. You can specify a UTC time in one of the following formats: <br/><ul><li>`YYYY-MM-DD` to specify a date. For example, `2020-11-29`.</li><li>`YYYY-MM-DDThh:mm:ss` to specify a time. For example, `2020-11-29T03:10:02`.</li></ul>
+|end |string |query |required |An end time (exclusive) in the UTC time zone before when the transactions will be fetched. You can specify a UTC time in one of the following formats: <br/><ul><li>`YYYY-MM-DD` to specify a date. For example, `2020-11-29`.</li><li>`YYYY-MM-DDThh:mm:ss` to specify a time. For example, `2020-11-29T03:10:02`.</li></ul>
+|includeTransactionType |string |query |optional |Which transaction types to fetch. <br>You can include more than one [supported transaction types](#supported-transaction-types) in a request. <br/>If you want to use `includeTransactionType`, don't use `limit` and `offset`. Because `limit` and `offset` will be ignored when `includeTransactionType` is used.  
+|limit |integer |query |optional |How many transactions to return in total in a response. You can specify `limit` with any integer greater than 0. <br/>If you want to use `limit` and `offset`, don't use `includeTransactionType`. Because `limit` and `offset` will be ignored when `includeTransactionType` is used. 
```

Comment:
>  don't use `limit` and `offset`. Because `limit` and `offset` will be ignored when `includeTransactionType` is used.  

This is not true. When i test with `includeTransactionType` and limit i get limited nr of transactions back.

Reply:
@igorleal could you check this finding?
As you mentioned before, in the code is that if the client passes in the query params "includeTransactionType", we ignore the limit and offset. They will be used only if the request came without the "includeTransactionType".

Reply:
So it looks like @iznatalia is right.

We have this:
```java
if (includeTransactionTypes.isEmpty()) {
            return accountTransactionDao.getAccountTransactionRange(
                accountIds,
                start,
                end,
                limit,
                offset
            );
        } else {
            return accountTransactionDao.getAccountTransactionRange(
                accountIds,
                start,
                end
            );
        }
```

But that's only some pre processing. When we are to actually get the account transactions we use it in both flows:

```java
if (!includeTransactionTypes.isEmpty()) {
                    return atDao.getAccountTransactions(
                        accountIds,
                        includeTransactionTypes,
                        start,
                        end,
                        minBktxId,
                        maxBktxId,
                        limit,
                        offset
                    );
                }
                return atDao.getAccountTransactions(
                    accountIds,
                    start,
                    end,
                    minBktxId,
                    maxBktxId,
                    limit,
                    offset
                );
```

Sorry for the mistake @Astro-Choco